### PR TITLE
Fixes HoP ticket messages not being visible unless on the ground

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

The messages alerting people that they were next in line or of a ticket's destruction were only visible when the tickets were on the ground. Not even the holder could see them. This fixes that. I'm not confident in the method though.

This also enables the HoP to delete the latest ticket, instead of having to bin it because he can't delete it until a new ticket is created that he can increment to.

## Why It's Good For The Game

Makes the thing work as intended.

## Changelog
:cl: cacogen
tweak: HoP can now delete the latest ticket
fix: HoP ticket messages should now be visible when the ticket is held
/:cl:

### CODE REVIEW

**Comment by actioninja** [comment]
that visible_message does not send when the item is inhand seems like more of a bug than intended functionality.

**Comment by PKPenguin321** [comment]
I think visible_message shoots the message to anything that shares a loc with it, with the loc in this case being the carbon mob holding it. I could kinda see that as a bug

**Comment by cacogen** [comment]
even the person holding it doesn't get the message though

**Comment by actioninja** [comment]
just going off the name of it and how it's expected to function I'm calling this 100% a bug.

**Comment by PKPenguin321** [comment]
>even the person holding it doesn't get the message though

Yeah, since the loc of the person is wherever they're standing, not inside themselves.

**Comment by cacogen** [comment]
#45997

**Comment by AnturK** [comment]
Should probably be a visible_message argument, it's not really obvious visible_message should be working like this but it's pretty common situation.

File: code/modules/paperwork/ticketmachine.dm [hunk 1]
```diff
@@ -48,17 +48,18 @@
...
-	say("Now serving ticket #[current_number]!")
-	if(!(obj_flags & EMAGGED) && tickets[current_number])
-		tickets[current_number].visible_message("<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
-	update_icon() //Update our icon here rather than when they take a ticket to show the current ticket number being served
+	if(current_number < ticket_number)
+		current_number ++ //Increment the one we're serving.
+		playsound(src, 'sound/misc/announce_dig.ogg', 50, 0)
+		say("Now serving ticket #[current_number]!")
+		if(!(obj_flags & EMAGGED) && tickets[current_number])
+			to_chat(viewers(3,get_turf(tickets[current_number].loc)), "<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
```
Comment (optimumtact) [inline code/modules/paperwork/ticketmachine.dm]: why not use visible_message on the ticket

Comment (optimumtact) [reply code/modules/paperwork/ticketmachine.dm]: er sorry I mean audible message

Comment (cacogen) [reply code/modules/paperwork/ticketmachine.dm]: @optimumtact works now, thanks

**Pull request merged**